src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -3735,6 +3739,11 @@ ImmutableArray<PropertySymbol> addProperties(ImmutableArray<ParameterSymbol> rec
                         {
                             // Deconstruct() is specified to simply assign from this property to the corresponding out parameter.
                             existingOrAddedMembers.Add(prop);
+
+                            if (isInherited)
+                            {
+                                validateNotHidden(prop, param);
```

Comment:
If `prop` is abstract, it won't be validated, is that intended? i.e:

```csharp
abstract record Base
{
    public abstract int I { get; init; }
}

record Derived(int I) : Base
{
    public int I() { return 0; }
}
```

Reply:
In that scenario `Derived` defined a concrete property `I` and a method `I()` (that's an error). The positional member is `Derived.I` so we didn't pick a hidden member, so the new error doesn't come into play.
Added tests to illustrate. Thanks for the scenario.